module/os/freebsd/zfs/zvol_os.c

```diff
@@ -825,7 +835,9 @@ zvol_cdev_write(struct cdev *dev, struct uio *uio, int ioflag)
 			dmu_tx_abort(tx);
 			break;
 		}
```

Comment:
There's an (unused) `uio_extflg` in uio. We could simply set a flag in the uio to indicate that the request is direct IO.

Reply:
The zvol code has been updated, but further debugging needs to happen with O_DIRECT.